src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -1376,5 +1385,14 @@ private void SyncFromBestHeightForRecoveredWallets(DateTime walletCreationDate)
                 this.walletSyncManager.SyncFromHeight(blockHeightToSyncFrom);
             }
         }
+
+        private TransactionItemModel FindSimilarReceivedTransactionOutput(List<TransactionItemModel> items, TransactionData transaction)
+        {
+            TransactionItemModel existingTransaction = items.FirstOrDefault(i => i.Id == transaction.Id &&
+                                                                                 i.Type == TransactionItemType.Received &&
+                                                                                 i.ConfirmedInBlock == transaction.BlockHeight &&
+                                                                                 i.Amount == transaction.Amount);
```

Comment:
Is it correct to compare the amounts too? If you have 3+ transaction outputs going to different addresses in the wallet, won't the third one onwards not match the amount as it accumulates within the `existingReceivedItem` in the caller?

Reply:
Removed amount for now. I thought that the main purpose was to sum amounts with the same value, but I guess it is ok for outputs with different value to be summed up.